resources/event-sources/templates/100-pre-upgrade-job.yaml

```diff
@@ -0,0 +1,136 @@
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: event-sources-pre-upgrade-helm
+rules:
+  - apiGroups: [""]
+    resources: ["pods", "pods/status"]
+    verbs: ["list", "get"]
+  - apiGroups: [""]
+    resources: ["pods/portforward"]
+    verbs: ["create", "get", "delete"]
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: event-sources-pre-upgrade-helm
+subjects:
+  - kind: ServiceAccount
+    name: event-sources-pre-upgrade
+    namespace: kyma-system
+roleRef:
+  kind: ClusterRole
+  name: event-sources-pre-upgrade-helm
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: event-sources-pre-upgrade-job
+  namespace: kyma-system
+  annotations:
+    helm.sh/hook: pre-upgrade
+    helm.sh/hook-weight: "10"
+spec:
+  backoffLimit: 1
+  template:
+    metadata:
+      name: event-sources-pre-upgrade-job
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: event-sources-pre-upgrade
+      restartPolicy: Never
+      containers:
+      - name: job
+        image: eu.gcr.io/kyma-project/test-infra/alpine-kubectl:v20200312-fc2a4147
+        terminationMessagePolicy: "FallbackToLogsOnError"
+        command:
+          - /bin/bash
+          - -c
+          - |
+            set -e
+            echo "*** event-sources-pre-upgrade-job starts ***"
+
+            echo "+ Deleting Event-Bus deployments"
+            kubectl -n kyma-system delete deployment --ignore-not-found \
+              event-bus-event-publish-service \
+              event-bus-subscription-controller
+
+            echo "+ Deleting Event-Bus SVCs"
+            kubectl -n kyma-system delete svc --ignore-not-found \
+              event-publish-service \
+              event-bus-event-publish-service-metrics \
+              event-bus-subscription-controller \
+              event-bus-subscription-controller-metrics
+
+            echo "+ Deleting Event-Bus Service Accounts"
+            kubectl -n kyma-system delete sa --ignore-not-found \
+              event-bus-event-publish-service \
+              event-bus-subscription-controller \
+              kyma-system event-bus-tests
+
+            echo "+ Deleting Event-Bus Helm package"
+            kubectl get -n kyma-installer secret helm-secret \
+              -o jsonpath="{.data['global\.helm\.ca\.crt']}" | base64 --decode > "$HOME/.helm/ca.pem"
+            kubectl get -n kyma-installer secret helm-secret \
+              -o jsonpath="{.data['global\.helm\.tls\.crt']}" | base64 --decode > "$HOME/.helm/cert.pem"
+            kubectl get -n kyma-installer secret helm-secret \
+              -o jsonpath="{.data['global\.helm\.tls\.key']}" | base64 --decode > "$HOME/.helm/key.pem"
+            if [[ $(helm list --tls event-bus) ]]; then
+              helm del --purge --tls event-bus
```

Comment:
Only to be sure, that if something is happened during the upgrade or if this job is running multiple time it will cleanup the resources. No error will be returned if nothing has to be removed. And it can be used with "ko" too.

Reply:
I think ko is not valuable here as the upgrade process is complex and there is no container creation involved here. The upgrade process is a [complex](https://kyma-project.io/docs/#installation-upgrade-kyma) one and helm upgrade and Kyma installer add complexity to it. Simulating with ko will not add value while development in this case.

Reply:
Only Helm is used now, but I like like to use ko during development phase, cause it reduce the time for a update, deploy, test significantly (from 20 minutes with Helm to 20 seconds...). And the pod generated bu ko contains all the functionality of "classic pod".   Of course, at the end, all should be tested inside of Helm environment.